test/data/BinaryHeap.ts

```diff
@@ -0,0 +1,740 @@
+          await instance['add(address)'](twoAddress);
+
+          expect(await instance['root()']()).to.equal(twoAddress);
+        });
+
+        describe('reverts if', function () {
+          it('index out of bounds', async () => {
+            await expect(instance['root()']()).to.be.revertedWith(
+              'BinaryHeap: index out of bounds',
+            );
+          });
+        });
+      });
+
+      describe('#add(address)', () => {
+        it('sets the parent node such that it is greater than or equal to the values of its children when a node is added', async () => {
+          await instance['add(address)'](twoAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](zeroAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](eightAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](threeAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](sixAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](fourAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](fiveAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](nineAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](sevenAddress);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(address)'](oneAddress);
+          checkNodes(await instance['toArray()']());
+        });
+
+        it('returns true if value is added', async () => {
+          expect(await instance.callStatic['add(address)'](zeroAddress)).to.be
+            .true;
+        });
+
+        it('returns false if value has already been added', async () => {
+          await instance.callStatic['remove(address)'](zeroAddress);
+          expect(await instance.callStatic['remove(address)'](zeroAddress)).to
+            .be.false;
+        });
+      });
+
+      describe('#remove(address)', () => {
+        it('sets the parent node such that it is greater than or equal to the values of its children when a node is removed', async () => {
+          await instance['add(address)'](twoAddress);
+          await instance['add(address)'](zeroAddress);
+          await instance['add(address)'](eightAddress);
+          await instance['add(address)'](threeAddress);
+          await instance['add(address)'](sixAddress);
+          await instance['add(address)'](fourAddress);
+          await instance['add(address)'](fiveAddress);
+          await instance['add(address)'](nineAddress);
+          await instance['add(address)'](sevenAddress);
+          await instance['add(address)'](oneAddress);
+
+          await instance['add(address)'](sixAddress);
```

Comment:
Should this be:
```suggestion
          await instance['remove(address)'](sixAddress);
```
?